### ISSUE TITLE
[REVIEW]: survxai: an R package for model agnostic explanations of survival models

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#961]
**Submitting author:** @AleksandraDabrowska (<a href="http://orcid.org/0000-0002-5963-6260">Aleksandra Grudziąż</a>)
**Repository:** <a href="https://github.com/MI2DataLab/survxai" target ="_blank">https://github.com/MI2DataLab/survxai</a>
**Version:** 0.2.0
**Editor:** @yochannah
**Reviewer:** @hiendn, @dirmeier
**Archive:** <a href="https://doi.org/10.5281/zenodo.1477857" target="_blank">10.5281/zenodo.1477857</a>

## Status

[![status](http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff/status.svg)](http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff"><img src="http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff/status.svg)](http://joss.theoj.org/papers/dcc9d53e8a1b1f613d59b9658b113fff)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/MI2DataLab/survxai" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@hiendn & @dirmeier, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @yochannah know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MI2DataLab/survxai">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (0.2.0)?
- [x] **Authorship:** Has the submitting author (@AleksandraDabrowska) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MI2DataLab/survxai">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (0.2.0)?
- [x] **Authorship:** Has the submitting author (@AleksandraDabrowska) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @hiendn, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by hiendn** [comment]
Hi @AleksandraDabrowska, I am one of your reviewers! My comments and suggestions will be posted on the issues page of your repository.

**Comment by yochannah** [comment]
Quick status check - looking at the original repo, issues [18](https://github.com/MI2DataLab/survxai/issues/18) to [21[(https://github.com/MI2DataLab/survxai/issues/21) need to be addressed before we can take this one forward, right?

**Comment by dirmeier** [comment]
Yeah, exactly.  From my side these two and [20](https://github.com/MI2DataLab/survxai/issues/20). All of them are minor changes.

**Comment by hiendn** [comment]
@yochannah that's correct, resolving [22](https://github.com/MI2DataLab/survxai/issues/22) would be good as well, if possible.

**Comment by dirmeier** [comment]
Hey @AleksandraDabrowska - could you rebuild the pdf once you are finished with the reviews? Then I can close the respective issues in your repository.

**Comment by AleksandraDabrowska** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.00961/joss.00961/10.21105.joss.00961.pdf)

**Comment by AleksandraDabrowska** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.00961/joss.00961/10.21105.joss.00961.pdf)

**Comment by hiendn** [comment]
I am now entirely happy with the with article.

**Comment by dirmeier** [comment]
Likewise. A a very nice package, too! :)

**Comment by AleksandraDabrowska** [comment]
@hiendn @dirmeier  Thank you very much for reviewing and for your kind words! :)
Should we now issue a DOI for our archive, as it is written in "The review process" section (https://joss.readthedocs.io/en/latest/submitting.html?fbclid=IwAR04SV6oxRM9B3mPL8NXhMhzToU_c3PwwcuiATwttKWaty2rEm8X8aZ19TM)?

**Comment by yochannah** [comment]
@AleksandraDabrowska Looking good! Before generating the archive, though, there is one issue with the manuscript I'd like to fix. Figures 3 and 4 are interspersed with the references in the most recent proof, but I'm reasonably sure they should be up in the main paper text. Can you fix this? Once this is fixed, please do generate an archive and share the URL in this thread!! 🎉

**Comment by AleksandraDabrowska** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.00961/joss.00961/10.21105.joss.00961.pdf)

**Comment by AleksandraDabrowska** [comment]
@yochannah here is archive https://zenodo.org/record/1477857#.W-AtQiDdg2w

**Comment by yochannah** [comment]
The figures look better! Thanks @AleksandraDabrowska

**Comment by yochannah** [comment]
@whedon set http://doi.org/10.5281/zenodo.1477857 as archive

**Comment by yochannah** [comment]
@arfon I think we're ready to accept this one!

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/48

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/48, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/49
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.00961](https://doi.org/10.21105/joss.00961)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@hiendn, @dirmeier - many thanks for your reviews here, and to @yochannah for editing this submission ✨ 

@AleksandraDabrowska - your paper is now accepted into JOSS :zap: :rocket: :boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00961/status.svg)](https://doi.org/10.21105/joss.00961)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.00961">
  <img src="http://joss.theoj.org/papers/10.21105/joss.00961/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.00961/status.svg
   :target: https://doi.org/10.21105/joss.00961
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00961/status.svg)](https://doi.org/10.21105/joss.00961)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
